plugins/packages/packages_inspector.rb

```diff
@@ -52,7 +68,49 @@ def inspect(_filter, _options = {})
     )
   end
 
-  def summary
-    "Found #{@description.packages.length} packages."
+  def inspect_dpkg
+    dpkg_data = @system.run_command(
+      "dpkg", "-l", stdout: :capture
+    )
+
+    lines = dpkg_data.lines.reject { |line| !line.start_with?("ii") }
+
+    packages = lines.map do |line|
+      name, version, arch = line.split[1..3]
+      version_segments = version.split("-")
+      release = version_segments.pop if version_segments.length > 1
+      version = version_segments.join("-")
+
+      Package.new(
+        name: name,
+        version: version,
+        release: release,
+        arch: arch
+      )
+    end
+
+    packages.each_slice(100) do |packages_slice|
+      apt_cache_output = @system.run_command(
+        "apt-cache",
+        "show",
+        *packages_slice.map { |p| "#{p.name}=#{[p.version, p.release].compact.join("-")}" },
+        stdout: :capture
+      )
+
+      packages_slice.each do |package|
+        name = Regexp.escape(package.name.sub(/:[^:]+$/, ""))
+        apt_cache_output =~
+          /Package: #{name}.*?MD5sum: (\w+).*?Origin: (\w+)/m
```

Comment:
I would add a newline or \s after `Package: #{name}` because there are packages like `zlib1g` and `zlib1g-dev` where the first would also match the second.